new-pipeline/src/active-weekly.js

```diff
@@ -0,0 +1,206 @@
+  "1.3": "Cheetah",
+  "1.4": "Puma",
+  "6": "Jaguar",
+  "7": "Panther",
+  "8": "Tiger",
+  "9": "Leopard",
+  "10": "Snow Leopard",
+  "11": "Lion",
+  "12": "Mountain Lion",
+  "13": "Mavericks",
+  "14": "Yosemite",
+  "15": "El Capitan",
+};
+
+
+var kCommaFormat = d3.format(',');
+var gFilters = {};
+for (let [name, sort] of kDimensionList) {
+  gFilters[name] = null;
+};
+
+var gWorker = new Worker("src-transformed/active-worker.js");
+gWorker.postMessage(["setup", kDimensionList]);
+gWorker.addEventListener("message",
+  function(e) {
+    switch (e.data[0]) {
+      case "fetch-complete":
+        getResults();
+        break;
+      case "results":
+        processResults(e.data[1]);
+        break;
+      default:
+        console.error("Unexpected event from worker", e);
+    }
+  }, false);
+
+$(document).ready(function() {
+  indicate("Fetching data...");
+  gWorker.postMessage(["fetch", "20151111"]);
+  d3.selectAll('.view-selector > tfoot')
+    .append('tr').append('th').append('input')
+    .attr('type', 'checkbox').classed('view-selectall', true).property('checked', true);
+  // d3 doesn't do delegated events well. Use jquery for that...
+  $(document).on("change", ".view-selectall", checkall_click);
+  $(document).on("click", ".view-selector > tbody > tr", filter_click);
+});
+
+function getResults() {
+  indicate("Calculating...");
+  gWorker.postMessage(["get-results", gFilters]);
+}
+
+function processResults(results) {
+  d3.select('#grand-total').text(kCommaFormat(results.grand_total));
+
+  d3.select('#sub-total').text(kCommaFormat(results.sub_total));
+  d3.select('#sub-total-pct').text(d3.format(".1%")(results.sub_total / results.grand_total));
+
+  let show_osversion = false;
+  let show_servicepack = false;
+
+  if (gFilters.os != null && gFilters.os.length == 1) {
+    show_osversion = true;
+    if (gFilters.osversion != null && gFilters.osversion.length == 1) {
+      show_servicepack = true;
+    }
+  }
+
+  d3.select("#osversion-note").classed("hidden", show_osversion);
+  d3.select("#osversion-view > .view-selector").classed("hidden", !show_osversion);
+  d3.select("#osservicepack-note").classed("hidden", show_servicepack);
+  d3.select("#osservicepack-view > .view-selector").classed("hidden", !show_servicepack);
+
+  for (let [name, sort] of kDimensionList) {
+    let ordered = results.dimensions[name];
+    if (sort == "key") {
+      ordered.sort(function(a, b) { return d3.ascending(a.key, b.key); });
+    } else {
+      ordered.sort(function(a, b) {
+        // descending by value, ascending by key
+        return b.value < a.value ? -1 : b.value > a.value ? 1 : a.key < b.key ? -1 : a.key > b.key ? 1 : 0;
+      });
+    }
+    // if (ordered.length > 10) {
+    //   var other = d3.sum(ordered.slice(10), function(d) { return d.value; });
+    //   ordered.splice(10, ordered.length - 10,
+    //     { key: "__other__", value: other });
+    // }
+
+    let keyFunction = function(d) { return d.key; };
```

Comment:
If you're using let, you could totally use arrow functions.
```js
let keyFunction = d => d.key;
```
(not a bug, not a nit... a fyi?)